packages/amplify_auth_cognito/lib/amplify_auth_cognito.dart

```diff
@@ -47,74 +47,75 @@ class AmplifyAuthCognito extends AuthPluginInterface {
     return streamWrapper.authStreamController;
   }
 
-  Future<SignUpResult> signUp({@required SignUpRequest request}) async {
+  Future<SignUpResult> signUp({required SignUpRequest request}) async {
     final res = await _instance.signUp(request: request);
     return res;
   }
 
   Future<SignUpResult> confirmSignUp(
-      {@required ConfirmSignUpRequest request}) async {
+      {required ConfirmSignUpRequest request}) async {
     final res = await _instance.confirmSignUp(request: request);
     return res;
   }
 
   Future<ResendSignUpCodeResult> resendSignUpCode(
-      {@required ResendSignUpCodeRequest request}) async {
+      {required ResendSignUpCodeRequest request}) async {
     final res = await _instance.resendSignUpCode(request: request);
     return res;
   }
 
-  Future<SignInResult> signIn({@required SignInRequest request}) async {
+  Future<SignInResult> signIn({required SignInRequest request}) async {
     final res = await _instance.signIn(request: request);
     return res;
   }
 
   Future<SignInResult> confirmSignIn(
-      {@required ConfirmSignInRequest request}) async {
+      {required ConfirmSignInRequest request}) async {
     final res = await _instance.confirmSignIn(request: request);
     return res;
   }
 
-  Future<SignOutResult> signOut({@required SignOutRequest request}) async {
+  Future<SignOutResult> signOut({SignOutRequest? request}) async {
     final res = await _instance.signOut(request: request);
     return res;
   }
 
   Future<UpdatePasswordResult> updatePassword(
-      {UpdatePasswordRequest request}) async {
+      {UpdatePasswordRequest? request}) async {
     final res = await _instance.updatePassword(request: request);
     return res;
   }
 
   Future<ResetPasswordResult> resetPassword(
-      {ResetPasswordRequest request}) async {
+      {ResetPasswordRequest? request}) async {
     final res = await _instance.resetPassword(request: request);
     return res;
   }
 
   Future<UpdatePasswordResult> confirmPassword(
-      {ConfirmPasswordRequest request}) async {
+      {ConfirmPasswordRequest? request}) async {
     final res = await _instance.confirmPassword(request: request);
     return res;
   }
 
-  Future<AuthUser> getCurrentUser({AuthUserRequest request}) async {
+  Future<AuthUser> getCurrentUser({AuthUserRequest? request}) async {
     final res = await _instance.getCurrentUser();
     return res;
   }
 
   Future<List<AuthUserAttribute>> fetchUserAttributes(
-      {AuthUserAttributeRequest request}) async {
+      {AuthUserAttributeRequest? request}) async {
```

Comment:
I think our initial thought was that the interface shoudl uniformally provide these so that when/if other, non-Cognito plugins are built (either by us or a customer) there is something already present.  But in retrospect this might be too much for some of these. However, before removing any entirely we should evaluate which of these methods might be give clientMetadata parameters to support Cognito Lambda Triggers.

Reply:
Alright, is that something we can do in another PR @haverchuck ?

Reply:
Yes.